luna.c

```diff
@@ -65,8 +65,8 @@ char *utf82unicode(char *in, char *end, unsigned long *c) {
 			*c |= *(in + 2) & 0b00111111;
 		return min(end, in + 3);
 	}
-	if ((*in & 0b111110000) == 0b11110000) {
-		*c = (*in & 0b00011111) << 18;
+	if ((*in & 0b11111000) == 0b11110000) {
+		*c = (*in & 0b00000111) << 18;
```

Comment:
For consistency, can you make it `*c = (*in & 0b00001111) << 12;` above?

Reply:
Well, it is consistent with the two bytes case (`*c = (*in & 0b00011111) << 6;` to get the lower 5 bits, not `*c = (*in & 0b00111111) << 6;`. Should I change that one too?

Reply:
In the three byte sequence case, only the lower four bits are part of the codepoint. It should go

```
0b00011111
0b00001111
0b00000111
```